lib/xapi/implementation.rb

```diff
@@ -9,35 +9,39 @@ class Implementation
       Regexp.new("\/\.$"),
     ]
 
-    attr_reader :track_id, :repo, :problem, :root
+    attr_reader :track_id, :repo, :problem, :root, :file_bundle
     attr_writer :files
     def initialize(track_id, repo, problem, root)
       @track_id = track_id
       @repo = repo
       @problem = problem
-      @root = root
+      @root = Pathname.new(root)
+      @file_bundle = FileBundle.new(dir, IGNORE)
     end
 
     def exists?
       File.exist?(dir)
     end
 
     def files
-      @files ||= Hash[paths.map {|path|
-        [filename(path), File.read(path)]
+      @files ||= Hash[file_bundle.list_paths.map {|path|
+        [path.relative_path_from(dir).to_s, File.read(path)]
       }].merge("README.md" => readme)
     end
 
     def zip
-      @zip ||= create_zip
```

Comment:
In this old case I called it `create_zip` because it was a private method, so I wasn't micromanaging.